login-utils/login.c

```diff
@@ -242,13 +242,10 @@ static void process_title_init(int argc, char **argv)
 	for (i = 0; envp[i] != NULL; i++)
 		continue;
 
-	environ = malloc(sizeof(char *) * (i + 1));
-	if (environ == NULL)
-		return;
+	environ = xmalloc(sizeof(char *) * (i + 1));
 
 	for (i = 0; envp[i] != NULL; i++)
-		if ((environ[i] = strdup(envp[i])) == NULL)
-			return;
+		environ[i] = xstrdup(envp[i]);
 	environ[i] = NULL;
```

Comment:
Thanks for this cleanup. I have moved it to login.c from lib/setproctitle.c without any change, but after your changes, it seems better.